relay/network_graph/dijkstra_weighted.py

```diff
@@ -103,7 +105,7 @@ def find_path(G, source, target, get_fee, value, max_hops=None, max_fees=None):
                 if n + 1 > max_hops:
                     continue
             if u in dist:
-                if vu_dist < dist[u]:
+                if n+1 < dist_hop[u]:  # will have to be updated in case we do a merged cost for hops and fees
```

Comment:
We discussed that we want to use the same logic like nodes are popped.

Reply:
I put a better logic for the test, removed the test on dist_hop for more visibility and because it logically cannot fail since we are using integer that cannot be negative and not costs functions.